Upstream-Status: Pending

# by default c++ include directories are not relative to "--sysroot"
# which brings one trouble when using the toolchain in an environment
# where the build directory generating that toolchain doesn't exist,
# e.g. in sstate, machine specific sysroot and relocatable SDK
# toolchain. This patch now enables c++ include paths under sysroot.
# This way it's enough as long as "--sysroot" is correctly enabled
# in the new environment.
#
# Signed-off-by Kevin Tian <kevin.tian@intel.com>, 2010-12-30

Index: gcc-4.6.0/gcc/cppdefault.c
===================================================================
--- gcc-4.6.0.orig/gcc/cppdefault.c
+++ gcc-4.6.0/gcc/cppdefault.c
@@ -48,15 +48,15 @@ const struct default_include cpp_include
 = {
 #ifdef GPLUSPLUS_INCLUDE_DIR
     /* Pick up GNU C++ generic include files.  */
-    { GPLUSPLUS_INCLUDE_DIR, "G++", 1, 1, 0, 0 },
+    { GPLUSPLUS_INCLUDE_DIR, "G++", 1, 1, 1, 0 },
 #endif
 #ifdef GPLUSPLUS_TOOL_INCLUDE_DIR
     /* Pick up GNU C++ target-dependent include files.  */
-    { GPLUSPLUS_TOOL_INCLUDE_DIR, "G++", 1, 1, 0, 1 },
+    { GPLUSPLUS_TOOL_INCLUDE_DIR, "G++", 1, 1, 1, 1 },
 #endif
 #ifdef GPLUSPLUS_BACKWARD_INCLUDE_DIR
     /* Pick up GNU C++ backward and deprecated include files.  */
-    { GPLUSPLUS_BACKWARD_INCLUDE_DIR, "G++", 1, 1, 0, 0 },
+    { GPLUSPLUS_BACKWARD_INCLUDE_DIR, "G++", 1, 1, 1, 0 },
 #endif
 #ifdef GCC_INCLUDE_DIR
     /* This is the dir for gcc's private headers.  */
